source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -436,21 +436,16 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const NodeGraph& n
     // Create output sockets from the nodegraph
     graph->addOutputSockets(nodeGraph);
 
-    if (context.getOptions().addUpstreamDependencies)
```

Comment:
Just removed this guard as it prevents internal nodes from being created. This version of the ShaderGraph::create function is used for creating nodegraph implementations where there are no external upstream connections, but the call to addUpstreamDependencies is used to create "internal" upstream dependencies (internal nodes upstream from the graphs outputs).

Reply:
addUpstreamDependencies was only set to true for OSL reference generation, so we had no check to catch this regression in our unittests.